signer/tests/integration/postgres.rs

```diff
@@ -1400,6 +1400,71 @@ async fn fetching_deposit_request_votes() {
     signer::testing::storage::drop_db(store).await;
 }
 
+#[cfg_attr(not(feature = "integration-tests"), ignore)]
+#[tokio::test]
+async fn fetching_deposit_signer_decisions() {
+    let pg_store = testing::storage::new_test_database().await;
+    let mut rng = rand::rngs::StdRng::seed_from_u64(51);
+
+    // This is just a sql test, where we use the `TestData` struct to help
+    // populate the database with test data. We set all the other
+    // unnecessary parameters to zero.
+    let num_signers = 3;
+    let test_model_params = testing::storage::model::Params {
+        num_bitcoin_blocks: 5,
+        num_stacks_blocks_per_bitcoin_block: 0,
+        num_deposit_requests_per_block: 1,
+        num_withdraw_requests_per_block: 0,
+        num_signers_per_request: num_signers,
+    };
+
+    let signer_set = testing::wsts::generate_signer_set_public_keys(&mut rng, num_signers);
+
+    let test_data = TestData::generate(&mut rng, &signer_set, &test_model_params);
+    test_data.write_to(&pg_store).await;
+
+    let chain_tip = test_data
+        .bitcoin_blocks
+        .iter()
+        .max_by_key(|x| (x.block_height, x.block_hash))
+        .unwrap();
+
+    for signer_pub_key in signer_set.iter() {
+        for deposit in test_data.deposit_requests.iter() {
+            let decision = model::DepositSigner {
+                txid: deposit.txid,
+                output_index: deposit.output_index,
+                signer_pub_key: *signer_pub_key,
+                can_accept: true,
+                can_sign: true,
+            };
+            pg_store
+                .write_deposit_signer_decision(&decision)
+                .await
+                .unwrap();
+        }
+    }
```

Comment:
Note that `TestData::generate` already [generates signers decision](https://github.com/stacks-network/sbtc/blob/main/signer/src/testing/storage/model.rs#L401-L412) (randomly) based on `num_signers_per_request`, and given that `write_deposit_signer_decision` `ON CONFLICT DO NOTHING` this snippet shouldn't have any effect (IIUC)